examples/ENS/recordsBundle.en.js

```diff
@@ -4,39 +4,41 @@ class Token {
 
     constructor(tokenInstance) {
         this.props = tokenInstance;
+        this.props.baseNode = ".eth";
+        this.props.fullName = this.props.ensName + this.props.baseNode;
+        let namehash = require("eth-ens-namehash");
+        web3.action.setProps({
```

Comment:
@James-Sangalli Seems better to move this into the `if` part of the if-else in `setup()` so it doesn't get called unnecessarily and risk creating an infinite loop.

Reply:
I disagree for 3 reasons:

1. AW is supposed to only resolve that once, it is unintuitive to have to think about an infinite loop here and it shouldn't happen anyway as AW protects against it.
2. The only way to resolve email record is to set the props and even then, the value could come back as empty (on count of the fact that there may be no email record at all) in which case the if statement will infinitely loop (although in reality it won't due to the protection of the wallet). 
3. The developer is not expected to know that set props calls a refresh, it makes sense to set such a thing in the constructor. 

What do you think?

Reply:
> 1. AW is supposed to only resolve that once, it is unintuitive to have to think about an infinite loop here and it shouldn't happen anyway as AW protects against it.

Yes, the TokenScript clients already protects against that now because the refresh (and thus `setProps()`) might happen before the attribute that is dependent on `nodeHash` (`emailRecord`) is evaluated.

> 2. The only way to resolve email record is to set the props and even then, the value could come back as empty (on count of the fact that there may be no email record at all)

Then maybe it's better to change it to just this:

```
if  (this.props.emailRecord == undefined) {
```

I know it wouldn't work now because Android returns "" by default rather than a nil/undefined. But I think the latter is a better behavior (but we can see if we want to fix this later, in a separate PR).

```
if (this.props.emailRecord == undefined) {
```

> 3. The developer is not expected to know that set props calls a refresh, it makes sense to set such a thing in the constructor.

I think this is the main point that matters most. I think in principle, we agree, we should lower the learning curve and provide the developer with a path of least surprise. So it does make sense to allow them to call `setProps()` everywhere. 

(I was also wondering if there's a situation where `setProps()` with the same value(s) will return a different result though. Imagine the attribute that is in question is a call to a price oracle, so the price may change despite the same input. But I suppose this is best resolved by how we allow the user to manage attribute lifetime/refreshing.)

So yeah, keep it :)

Reply:
Yup, I just put two conditions in the if statement instead, it's a shame that a === is not safe because undefined can also be 'undefined' (both a string and it's own type)